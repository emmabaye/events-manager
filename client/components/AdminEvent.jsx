import React, { Component } from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { cancelCenterEvent } from '../actions/centerAction';

/**
 * React component for admin event
 * card displayed in the admin panel
 *
 * @return {undefined}
 */
class AdminEvent extends Component {
  cancelCenterEvent = () => {
    console.log("HELLOP");
    let { id } = this.props.event;
    this.props.dispatch(cancelCenterEvent(id));
  }

  /**
   * React's method to render react component
   * @return {object}
   */
  render() {
    return (
      <div className="col-md-4 event">
        <div className="card">
          <div className="card-block">
            <h5 className="card-title"><b>{this.props.event.title}</b></h5>
            <p className="card-text h6">{this.props.event.description}</p>
            <p className="card-text h6"><b>Venue</b>: {this.props.event.venue}</p>
            <p className="card-text h6"><b>Date</b>: {moment(this.props.event.date).format('DD MMMM YYYY')}</p>
            <p className="card-text h6"><b>Time</b>: { this.props.event.time }</p>
            <p className="card-text h6"><b>Status</b>: {this.props.event.status}</p>
            <p className="text-center">
              <Link to={`/events/${this.props.event.id}`} className="btn btn-sm btn-primary">
                <i className="fa fa-info-circle fa-lg" /> View Event
              </Link>
              {this.props.event.status !== 'cancelled' &&
                <a href="#dcdf" onClick={this.cancelCenterEvent} className="btn btn-sm btn-danger" >
                  <i className="fa fa-times fa-lg" /> Cancel
                </a>
              }
            </p>

          </div>
        </div>
      </div>

    );
  }
}

export default AdminEvent;
