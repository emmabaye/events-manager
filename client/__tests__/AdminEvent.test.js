import { BrowserRouter } from 'react-router-dom';
import AdminEvent from '../components/AdminEvent.jsx';

describe('AdminEvent Component', () => {
  const props = {
    event: {
      id: "1",
      title: "",
      description: "",
      location: "",
      date: "",
      time: "",
      status: 'true'
    },
    show: () => {}
  };

  it('it should render component shallow', () => {
    const wrapper = shallow(<BrowserRouter><AdminEvent {...props} /></BrowserRouter>);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render AdminEvent component mounted', () => {
    const wrapper = mount(<BrowserRouter><AdminEvent {...props} /></BrowserRouter>);
    expect(wrapper.length).toEqual(1);
  });

 /*
  it('it should handle onClick event', () => {
    const wrapper = shallow(<BrowserRouter><AdminEvent {...props} /></BrowserRouter>);
    let viewEventButton = wrapper.find('Link');
    viewEventButton.simulate('click');
  });
  */
  
});
