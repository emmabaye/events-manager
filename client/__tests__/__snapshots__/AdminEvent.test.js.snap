// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminEvent Component it should render component shallow 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <AdminEvent
    event={
      Object {
        "date": "",
        "description": "",
        "id": "1",
        "location": "",
        "status": "true",
        "time": "",
        "title": "",
      }
    }
    show={[Function]}
  />
</Router>
`;
